client/my-sites/people/people-invite-details/index.jsx

```diff
@@ -19,17 +19,33 @@ import HeaderCake from 'components/header-cake';
 import Card from 'components/card';
 import PeopleListItem from 'my-sites/people/people-list-item';
 import Gravatar from 'components/gravatar';
+import Button from 'components/button';
 import QuerySiteInvites from 'components/data/query-site-invites';
 import EmptyContent from 'components/empty-content';
 import { getSelectedSite } from 'state/ui/selectors';
-import { isRequestingInvitesForSite, getInviteForSite } from 'state/invites/selectors';
+import {
+	isRequestingInvitesForSite,
+	getInviteForSite,
+	isDeleting,
+	didDeletionSucceed,
+} from 'state/invites/selectors';
+import { deleteInvite } from 'state/invites/actions';
 
 class PeopleInviteDetails extends React.PureComponent {
 	static propTypes = {
 		site: PropTypes.object,
 		inviteKey: PropTypes.string.isRequired,
 	};
 
+	componentWillReceiveProps( nextProps ) {
+		if (
+			true === nextProps.deleteSuccess &&
+			nextProps.deleteSuccess !== this.props.deleteSuccess
+		) {
+			this.goBack();
+		}
```

Comment:
I have a couple of concerns about this approach:

- It feels fragile.  It relies on an action being dispatched from elsewhere in this component, then afterwards, some Redux state needs to be updated and a selector needs to change its return to a specific value in order to trigger this logic.  I can't think of a better approach off-hand, and I'd feel better about it if we had a test covering the intended behavior.
- It briefly displays the "The requested invite does not exist" message after the deletion succeeds and before the `this.goBack()` takes effect.  If we keep the current approach, we should avoid the re-render in this state (placeholder is probably a good alternative, since this moment is also a loading state).